src/main/java/logic/AdvancementLogic.java

```diff
@@ -0,0 +1,111 @@
+
+    @Override
+    public void onEvent(SlashCommandInteraction event, User user, VelenArguments args, List<SlashCommandInteractionOption> options, InteractionImmediateResponseBuilder firstResponder) {
+        final SlashCommandInteractionOption subcommandOption = event.getOptions().get(0);
+        final Optional<Long> targetFacets = subcommandOption.getOptionLongValueByName("target-facets");
+        if (targetFacets.isPresent()) {
+            final Boolean expert = subcommandOption.getOptionBooleanValueByName("target-expert").orElse(false);
+            final List<Integer> startingArray = subcommandOption.getOptionStringValueByName("starting-array")
+                    .map(s -> Arrays.stream(s.split(" ")).map(Integer::parseInt).collect(Collectors.toList()))
+                    .orElse(new ArrayList<>());
+            final long minimumFacets = subcommandOption.getOptionLongValueByName("minimum-facets").orElse(12L);
+            final boolean tall = subcommandOption.getName().equals("tall");
+            final Boolean adolescentInterests = subcommandOption.getOptionBooleanValueByName("adolescent-interests").orElse(false);
+            final long nervewrightMana = subcommandOption.getOptionLongValueByName("nervewright-mana").orElse(0L);
+            final boolean nonEphemeral = subcommandOption.getOptionBooleanValueByName("non-ephemeral").orElse(false);
+
+            final Long targetFacetsActual = targetFacets.get();
+            List<SpecialtySkill> skills = new GeneralSkillCalculator(startingArray, expert, tall, minimumFacets, adolescentInterests, nervewrightMana).generate(targetFacetsActual);
+            EmbedBuilder resultsEmbed = new EmbedBuilder();
+            resultsEmbed.setTitle("How to get this general skill to f" + targetFacetsActual);
+            resultsEmbed.setDescription("In order to advance this general skill and go " + subcommandOption.getName() + ", you can raise the specialty skills under the general skill like this (assuming you have a high enough attribute or mentors):");
+            final Optional<String> existingSpecialtyAdvancement = skills.stream()
+                    .filter(specialtySkill -> specialtySkill.getStartingFacets() != 0)
+                    .map(specialtySkill -> MessageFormat.format("f{0} → f{1} ({2} AP)", specialtySkill.getStartingFacets(), specialtySkill.getCurrentFacets(), specialtySkill.getAPSpent()))
+                    .reduce((s, s2) -> MessageFormat.format("{0}\n{1}", s, s2));
+            existingSpecialtyAdvancement.ifPresent(s -> resultsEmbed.addInlineField("Existing Specialties", s));
+            final Optional<String> newSpecialtySkillAdvancement = skills.stream()
+                    .filter(specialtySkill -> specialtySkill.getStartingFacets() == 0)
+                    .map(specialtySkill -> MessageFormat.format("f{0} ({1} AP)", specialtySkill.getCurrentFacets(), specialtySkill.getAPSpent()))
+                    .reduce((s, s2) -> MessageFormat.format("{0}\n{1}", s, s2));
```

Comment:
Just use one of the String collectors?

Reply:
Switched to `.collect(Collectors.joining("\n"))`